tests/conftest.py

```diff
@@ -149,21 +151,23 @@ def erepo(repo_dir):
         repo.dvc.scm.commit("init repo")
 
         rconfig = RemoteConfig(repo.dvc.config)
-        rconfig.add("upstream", repo.dvc.cache.local.cache_dir, default=True)
+        rconfig.add("upstream", repo.upstream_path, default=True)
         repo.dvc.scm.add([repo.dvc.config.config_file])
         repo.dvc.scm.commit("add remote")
 
         repo.create("version", "master")
         repo.dvc.add("version")
         repo.dvc.scm.add([".gitignore", "version.dvc"])
         repo.dvc.scm.commit("master")
+        repo.dvc.push()
```

Comment:
Why is this needed? And all upstream_path stuff in general? Sorry, might've missed some discussion somewhere.

Reply:
It seems to me that erepo had default remote configured pointing to its own cache, in order to not trigger infamous NoRemoteError on any test related to `get` and `import`. In new test `test_from_non_dvc_,aster`, where I am removing `.dvc` I had to either manually specify new storage for `erepo`, or adjust `erepo` fixture. I choose the latter because I believe that setting cache as default remote is not realistic use case, which also omits some important details (eg you don't need to push to "add" something to remote cache, because its local cache and adding is done automatically) so we were testing some very specific use cases everywhere where we used `erepo`.

Reply:
@pared But you are not even using "upstream_path" in your tests. And you have to modify a bunch of unrelated code for no good reason. In your test you simply need `.dvc` to not be tracked on master, which you could do by simply adding `.dvc` to gitignore on master and commiting.

Reply:
The point about it being not realistic is a good one, but we will soon do that automatically by-default  https://github.com/iterative/dvc/issues/2599 anyway, and, well, it is not nice to break other tests because of 1 test for specific use case.

Reply:
Ok, sorry I didn't noticed that I am modifying repo, its leftover from some previous changes.
Still, I believe this change is desirable. I can achieve the desired `master` state with properly editing `.gitignore` but the actual problem is that erepo is not an actual use case. For example, `test_get_repo_file` should fail in original `erepo` setup. We were using a trick to make it pass, but without the push that we are commenting on, it should not pass.

Reply:
> The point about it being not realistic is a good one, but we will soon do that automatically by-default #2599 anyway, and, well, it is not nice to break other tests because of 1 test for specific use case.

But setting default remote to be a cache is a trick, so in my opinion, it's not breaking other tests, its actually fixing them, since they were not actual use case.

Reply:
#2599 should fix our discussion, but since it's not in the current sprint, I wouldn't just leave it be, especially that this change is not that big.

Reply:
@pared Sorry, I don't get the point about `test_get_repo_file`, don't see any tricks in it.
```
def test_get_repo_file(erepo):                                                  
    src = erepo.FOO                                                             
    dst = erepo.FOO + "_imported"                                               
                                                                                
    Repo.get(erepo.root_dir, src, dst)                                          
                                                                                
    assert os.path.exists(dst)                                                  
    assert os.path.isfile(dst)                                                  
    assert filecmp.cmp(erepo.FOO, dst, shallow=False)  
```
My point is that you don't need to modify `erepo` and stuff around for your 1 specific test, you can do it with 2 lines that will add `.dvc` dir to gitignore and commit it on master.

Reply:
There is no trick in the test, the trick is inside `erepo`:
```
rconfig = RemoteConfig(repo.dvc.config)
rconfig.add("upstream", repo.dvc.cache.local.cache_dir, default=True)
```

It's not about modifying whole erepo for one test. I believe `erepo` is not being set up properly for tests.

Reply:
@pared Ah, got it. But that trick is something that we will implement in https://github.com/iterative/dvc/issues/2599 by default, and it doesn't seem to break any other tests, it is just an internal implementation detail for erepo. But I also understand your point about it being hacky. Maybe let's create an issue for it and for now stick with gitignore, so that we don't have to touch all other tests? It doesn't seem like it is in the scope of this PR.

Reply:
#2888